### PR TITLE
Remove support for key based auth - azure open ai and azure ai search

### PR DESCRIPTION
**Pull request by manvkaur**
Addresses #48 
- Removed Azure OpenAI Key based auth support
- Removed Azure AI Search, Search API Key based auth support
- Added more clarity in README around roles to be assigned
- Samples updated

### CODE REVIEW

File: tests/SampleValidation/Chat.cs [hunk 1]
```diff
@@ -32,6 +32,9 @@ public async Task SuperbowlQuestions()
         string baseAddress = Environment.GetEnvironmentVariable("FUNC_BASE_ADDRESS") ?? "http://localhost:7071";
         string chatId = $"superbowl-{Guid.NewGuid():N}";
 
+        string functionKey = Environment.GetEnvironmentVariable("FUNC_KEY") ?? throw new InvalidOperationException("Missing environment variable 'FUNC_KEY'");
```
Comment (cgillum) [inline tests/SampleValidation/Chat.cs:35]: IIRC, Functions has three types of keys, API keys, system keys, and a master key. Which type of key does this environment variable represent? Should we update the environment variable name to make this more clear?

File: build/build-pr-ci.yml [hunk 2]
```diff
@@ -64,24 +41,50 @@ steps:
...
+    inputs:
+      connectedServiceNameARM: $(serviceConnectionName)
+      appType: 'functionApp'
+      appName: $(functionAppName)
+      package: '$(build.artifactstagingdirectory)/csharp-ooproc.zip'
+      deploymentMethod: 'auto'
+
   - script: dotnet test --configuration $(config) --no-build --verbosity normal
+    env:  
+      FUNC_KEY: $(FUNC_KEY)
```
Comment (cgillum) [inline build/build-pr-ci.yml:81]: Where is this defined? Can we add a comment explaining what this is for and where it is configured?

File: README.md [hunk 3]
```diff
@@ -260,7 +261,9 @@ public async Task GenerateEmbeddings_Http_RequestAsync(
 
 The semantic search feature allows you to import documents into a vector database using an output binding and query the documents in that database using an input binding. For example, you can have a function that imports documents into a vector database and another function that issues queries to OpenAI using content stored in the vector database as context (also known as the Retrieval Augmented Generation, or RAG technique).
 
- The supported list of vector databases is extensible, and more can be added by authoring a specially crafted NuGet package. Visit the currently supported vector specific folder for specific usage information:
+Only raw text and file types with extension `.txt` are supported at present for data ingestion.
```
[the remark this answers is not in the document]
Comment (manvkaur) [reply README.md:262]: yes, that's correct. we support raw text, updated to just mention that

File: build/build-pr-ci.yml (hunk 2 above)
Comment (manvkaur) [reply build/build-pr-ci.yml:81]: this is defined in pipeline variables and value is kept as secret, added a comment for the same.

File: tests/SampleValidation/Chat.cs (hunk 1 above)
Comment (manvkaur) [reply tests/SampleValidation/Chat.cs]: Its the default function key (also known as a host key), I have updated the variable name to FUNC_DEFAULT_KEY.
Let me know if it is good.

**Review by cgillum — APPROVED** [review]

**Pull request merged**